malduck/crypto/rsa.py

```diff
@@ -20,25 +20,27 @@ class PublicKeyBlob(BaseBlob):
 
     def __init__(self) -> None:
         BaseBlob.__init__(self)
-        self.e = None
-        self.n = None
+        self.e: Optional[int] = None
+        self.n: Optional[int] = None
 
-    def parse(self, buf: BytesIO) -> int:
+    def parse(self, buf: BytesIO) -> Optional[int]:
         header = buf.read(12)
         if len(header) != 12 or header[:4] != self.magic:
-            return
+            return None
 
-        self.bitsize = uint32(header[4:8])
-        self.e = int(uint32(header[8:12]))
+        self.bitsize = cast(int, uint32(header[4:8], fixed=False))
+        self.e = cast(int, uint32(header[8:12], fixed=False))
 
         n = buf.read(self.bitsize // 8)
         if len(n) != self.bitsize // 8:
-            return
+            return None
 
         self.n = bigint.unpack(n)
         return 12 + self.bitsize // 8
 
     def export_key(self) -> bytes:
+        if not (self.e and self.n):
+            raise Exception("The imported key is invalid")
```

Comment:
```suggestion
            raise ValueError("The imported key is invalid")
```

IMO invalid class state should be considered as `invalid argument` situation. I agree with StackOverflow: https://stackoverflow.com/questions/10726919/what-error-to-raise-when-class-state-is-invalid